ObjC build warnings

```
make: *** [test/ios/controller[iphoneos-arc-xcodesdk]] Error 65
/Users/joe/Developer/zeroc-ice/ice/objective-c/test/Common/TestCommon.m:214:11: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
  214 | serverStop()
      |           ^
      |            void
1 warning generated.
/Users/joe/Developer/zeroc-ice/ice/objective-c/test/Common/TestCommon.m:214:11: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
  214 | serverStop()
      |           ^
      |            void
1 warning generated.
/Users/joe/Developer/zeroc-ice/ice/objective-c/test/Ice/ami/AllTests.m:807:16: warning: variable 'r' may be uninitialized when used here [-Wconditional-uninitialized]
  807 |         test(![r isSent]);
      |                ^
In file included from /Users/joe/Developer/zeroc-ice/ice/objective-c/test/Ice/ami/AllTests.m:6:
../../include/TestCommon.h:43:20: note: expanded from macro 'test'
   43 | #define test(ex) ((ex) ? ((void)0) : testFailed(#ex, __FILE__, __LINE__))
      |                    ^~
/Users/joe/Developer/zeroc-ice/ice/objective-c/test/Ice/ami/AllTests.m:800:29: note: initialize the variable 'r' to silence this warning
  800 |         id<ICEAsyncResult> r;
      |                             ^
      |                              = nil
1 warning generated.
```

--- a/objective-c/test/Common/TestCommon.m
+++ b/objective-c/test/Common/TestCommon.m
@@ -211,7 +211,7 @@ serverReady(id<ICECommunicator> c)
 }
 
 void
-serverStop()
+serverStop(void)
 {
     @try
     {

--- a/objective-c/test/Ice/ami/AllTests.m
+++ b/objective-c/test/Ice/ami/AllTests.m
@@ -797,7 +797,7 @@ amiAllTests(id<ICECommunicator> communicator, BOOL collocated)
     {
         [testController holdAdapter];
 
-        id<ICEAsyncResult> r;
+        id<ICEAsyncResult> r = nil;
         ICEByte buf[10024];
         ICEByteSeq* seq = [ICEByteSeq dataWithBytes:buf length:sizeof(buf)];
         for(int i = 0; i < 200; ++i) // 2MB

